packages/backend/src/apps/gathersg/actions/update-case/index.ts

```diff
@@ -0,0 +1,180 @@
+import type { IRawAction } from '@plumber/types'
+
+import { ZodError } from 'zod'
+import { fromZodError } from 'zod-validation-error'
+
+import HttpError from '@/errors/http'
+import StepError, { GenericSolution } from '@/errors/step'
+import { ensureZodEnumValue } from '@/helpers/zod-utils'
+
+import { GatherSGError } from '../../common/types'
+
+import { fieldTypeEnum, requestSchema, responseSchema } from './schema'
+
+const action: IRawAction = {
+  name: 'Update case',
+  key: 'updateCase',
+  description: 'Update a case based on the case id (UUID)',
+  arguments: [
+    {
+      label: 'Case ID',
+      key: 'caseId',
+      type: 'string' as const,
+      description:
+        'You can only select a step variable here. You should be using a Tile to store your case IDs to make reference to.',
+      required: true,
+      variables: true,
+      singleVariableSelection: true,
+    },
+    // TODO: see if it is possible to get all possible statuses from the API
+    {
+      label: 'Case status',
+      key: 'caseStatus',
+      type: 'string' as const,
+      description: 'Enter the status you want to update the case to.',
+      required: true,
+      variables: true,
+      hiddenIf: {
+        fieldKey: 'caseId',
+        op: 'is_empty',
+      },
+    },
+    {
+      label: 'Case fields',
+      key: 'caseFields',
+      type: 'multirow-multicol' as const,
+      required: true,
+      description:
+        'Specify values for each field you want to update in your case. Note that fields that require an array of objects as a value are not supported yet.',
+
+      subFields: [
+        {
+          placeholder: 'Field',
+          key: 'field',
+          type: 'dropdown' as const,
+          showOptionValue: false,
+          required: true,
+          variables: false,
+          source: {
```

Comment:
i think can `allowArbitrary` here so that users can add new fields if not present in the case

Reply:
nice, shall do so thanks